test/controllers/actions/AuthActionSpec.scala

```diff
@@ -35,8 +35,76 @@ class AuthActionSpec extends SpecBase {
     def onPageLoad() = authAction { _ => Results.Ok }
   }
 
+  private def authRetrievals(affinityGroup: AffinityGroup = AffinityGroup.Individual, enrolment: Enrolments = noEnrollment) =
+    Future.successful(new ~(new ~(Some("id"), Some(affinityGroup)), enrolment))
+
+  private val agentAffinityGroup = AffinityGroup.Agent
+  private val orgAffinityGroup = AffinityGroup.Organisation
+  private val noEnrollment = Enrolments(Set())
+  private val agentEnrolment = Enrolments(Set(Enrolment("HMRC-AS-AGENT",List(EnrolmentIdentifier("SomeKey","SomeVal")),"Activated",None)))
+
+
   "Auth Action" when {
 
+
+    "Agent user hasn't enrolled in Trusts" must {
+
+      "redirect the user to the create agent services page" in {
+
+        val application = applicationBuilder(userAnswers = None).build()
+        val bodyParsers = application.injector.instanceOf[BodyParsers.Default]
+        val authAction = new AuthenticatedIdentifierAction(new FakeAuthConnector(authRetrievals(agentAffinityGroup, noEnrollment)), frontendAppConfig, bodyParsers)
+        val controller = new Harness(authAction)
+        val result = controller.onPageLoad()(fakeRequest)
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(controllers.routes.CreateAgentServicesAccountController.onPageLoad().url)
+      }
+
+    }
+
+    "Agent user has correct enrolled in Trusts" must {
```

Comment:
@jassalrichy Minor change in that the Agent has enrolled for AgentServicesAccount not Trusts.